runtime/nsd_ossl.c

```diff
@@ -1710,13 +1756,17 @@ Connect(nsd_t *pNsd, int family, uchar *port, uchar *host, char *device)
 		ABORT_FINALIZE(RS_RET_NO_ERRCODE);
 	}
 
+	CHKiRet(SetServerNameIfPresent(pThis, host));
```

Comment:
Question: Why does this need to be checked if if "pThis->remoteSNI == NULL?
Like in nsd_gtls.c line 2022.

Reply:
This is actually a slight change to `nsd_ossl`, which currently behaves differently than `nsd_gtls` (if I'm reading the code right.) The change made in PR #1393 to set the SNI based on the hostname, was only made to `nsd_gtls` - so currently, `nsd_ossl` actually has no SNI support at all. I'll update the PR to indicate that this is actually also a feature add to `nsd_ossl`.

But if I'm understanding your question right: the reason to call `SetServerNameIfPresent` even if `pThis->remoteSNI == NULL`, is to set a default SNI (the value of the `target` parameter to `omfwd`, if it's not an IP address) if no SNI is specified in configuration.

Reply:
Thanks for the explanation (Y).